code/modules/reagents/reagent_containers/cups/_cup.dm

```diff
@@ -19,6 +19,17 @@
 	///Whether the 'bottle' is made of glass or not so that milk cartons dont shatter when someone gets hit by it.
 	var/isGlass = FALSE
 
+/obj/item/reagent_containers/cup/dropped(mob/user)
+	. = ..()
+	reagents.flags = initial(reagent_flags)
+	if(reagent_container_liquid_sound && reagents.total_volume > 0)
+		playsound(src, reagent_container_liquid_sound, LIQUID_SLOSHING_SOUND_VOLUME, vary = TRUE, ignore_walls = FALSE)
```

Comment:
if this behavior is only meant to be done by cups, then i think u should remove the reagent_container_liquid_sound var from the base type since its gonna be unused there